Mathlib/CategoryTheory/Bicategory/Adjunction/Mate.lean

```diff
@@ -214,6 +215,371 @@ theorem mateEquiv_hcomp (α : g ≫ l₂ ⟶ l₁ ≫ h) (β : h ≫ l₄ ⟶ l
 
 end mateEquivHComp
 
+section mateEquivSquareComp
+
+variable {a b c d e f x y z : B}
+variable {g₁ : a ⟶ d} {h₁ : b ⟶ e} {k₁ : c ⟶ f} {g₂ : d ⟶ x} {h₂ : e ⟶ y} {k₂ : f ⟶ z}
+variable {l₁ : a ⟶ b} {r₁ : b ⟶ a} {l₂ : b ⟶ c} {r₂ : c ⟶ b} {l₃ : d ⟶ e} {r₃ : e ⟶ d}
+variable {l₄ : e ⟶ f} {r₄ : f ⟶ e} {l₅ : x ⟶ y} {r₅ : y ⟶ x} {l₆ : y ⟶ z} {r₆ : z ⟶ y}
+variable (adj₁ : l₁ ⊣ r₁) (adj₂ : l₂ ⊣ r₂) (adj₃ : l₃ ⊣ r₃)
+variable (adj₄ : l₄ ⊣ r₄) (adj₅ : l₅ ⊣ r₅) (adj₆ : l₆ ⊣ r₆)
+
+/-- Squares of squares between left adjoints can be composed by iterating vertical and horizontal
+composition.
+-/
+def leftAdjointSquare.comp
```

Comment:
Could you enclose certain definitions in sections when they have the same variables, like the first three definitions here?

Reply:
Sorry for the late response. I've made sections for these lemmas.